hydra-node/src/Hydra/Options.hs

```diff
@@ -550,6 +557,40 @@ hydraNodeCommand =
       (decodeUtf8 $ encodePretty Contract.scriptInfo)
       (long "script-info" <> help "Dump script info as JSON")
 
+defaultContestationPeriod :: ContestationPeriod
+defaultContestationPeriod = UnsafeContestationPeriod 3
+
+contestationPeriodParser :: Parser ContestationPeriod
+contestationPeriodParser =
+  option
+    contestationParser
+    ( long "contestation-period"
+        <> metavar "CONTESTATION-PERIOD"
+        <> value defaultContestationPeriod
+        <> showDefault
+        <> completer (listCompleter ["60", "180", "300"])
+        <> help
+          "Contestation period for close transaction in seconds. \
+          \ If this value is not in sync with other participants hydra-node will ignore the initial tx.\
+          \ Additionally, this value needs to make sense compared to the current network we are running."
+    )
+ where
+  -- Parses formats like "60" and "60s" interchangeably.
+  -- Provided argument needs to be higher than zero.
+  contestationParser :: ReadM ContestationPeriod
+  contestationParser = eitherReader $ \s -> do
+    let errorMsg = "Malformed argument provided for contestation-period: " <> s
+    eparsed <- maybe (Left errorMsg) (pure . Right) $ do
+      cpArgStr <-
+        case viaNonEmpty last s of
+          Just 's' -> viaNonEmpty init s
+          _ -> Just s
+      readMaybe cpArgStr
+    case eparsed of
+      Right 0 -> Left "contestation-period argument needs to be higher than zero"
+      Right a -> Right $ UnsafeContestationPeriod a
+      Left _ -> Left errorMsg
```

Comment:
How about: `parseNatural <|> parseNominalDiffTime` with `parseNatural = UnsafeContestationPeriod <$> auto` and

```
parseNominalDiffTime = auto >>= \diffTime -> do
  let s = nominalDiffTimeToSeconds diffTime
  if s <= 0
  then fail "negative contestation period"
  else UnsafeContestationPeriod $ fromIntegral s
```
?

Reply:
(I usually think in combining cases with backtracing, rather scrutinizing for which case it is -> looking for the `'s'`)

Reply:
yes, looks a lot nicer